core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -268,6 +270,28 @@ object UnitAutomation {
         return true
     }
 
+    private fun tryTakeBackCapturedCity(unit: MapUnit): Boolean {
+        var capturedCities = unit.civInfo.gameInfo.civilizations
+                .filter { unit.civInfo.isAtWarWith(it) }
+                .flatMap { it.cities }.asSequence()
+                .filter { unit.civInfo.civName == it.foundingCiv && it.isInResistance() && it.health < it.getMaxHealth() } //Most likely just been captured
+                .asSequence()
+
+        if (unit.type.isRanged()) // ranged units don't harm capturable cities, waste of a turn
+            capturedCities = capturedCities.filterNot { it.health == 1 }
+
+        val closestReachableCapturedCity = capturedCities
+                .asSequence().map { it.getCenterTile() }
+                .filter { unit.movement.canReach(it) }
```

Comment:
.sortedBy {it.aerialDistanceTo(unit.getTile())} 
.firstOrNull{ unit.movement.canReach(it) }